prompt2model_demo.ipynb

```diff
@@ -0,0 +1,550 @@
+    "test_datasets = [each[\"test\"] for each in text_modified_dataset_dicts]\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Combine the retrieved dataset with generated dataset and use the `GenerationModelTrainer` to finetune the retrieved model. After the finetuning, we save the model and tokenizer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from prompt2model.model_trainer import GenerationModelTrainer\n",
+    "from pathlib import Path\n",
+    "\n",
+    "trainer = GenerationModelTrainer(\n",
+    "    pre_train_model_name,\n",
+    "    has_encoder=True,\n",
+    "    executor_batch_size=1,\n",
+    "    tokenizer_max_length=1024,\n",
+    "    sequence_max_length=1280,\n",
+    ")\n",
+    "\n",
+    "args_output_root = Path(\"result/training_output\")\n",
+    "args_output_root.mkdir(parents=True, exist_ok=True)\n",
+    "\n",
+    "trained_model, trained_tokenizer = trainer.train_model(\n",
+    "    hyperparameter_choices={\n",
+    "        \"output_dir\": str(args_output_root),\n",
+    "        \"save_strategy\": \"epoch\",\n",
+    "        \"num_train_epochs\": 1,\n",
+    "        \"per_device_train_batch_size\": 8,\n",
+    "        \"evaluation_strategy\": \"epoch\",\n",
+    "    },\n",
+    "    training_datasets=train_datasets,\n",
+    "    validation_datasets=val_datasets,\n",
+    ")\n",
+    "\n",
+    "trained_model.save_pretrained(\"trained_model\")\n",
+    "trained_tokenizer.save_pretrained(\"trained_tokenizer\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Try it out!\n",
+    "\n",
+    "Now, you can add input and use your fine-tuned model to do inference."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from prompt2model.model_executor import GenerationModelExecutor\n",
+    "\n",
+    "model_executor = GenerationModelExecutor(trained_model, trained_tokenizer)\n",
+    "# CHANGE THIS to your own input\n",
+    "input = \"How many departments are within the Stinson-Remick Hall of Engineering? The College of Engineering was established in 1920, however, early courses in civil and mechanical engineering were a part of the College of Science since the 1870s. Today the college, housed in the Fitzpatrick, Cushing, and Stinson-Remick Halls of Engineering, includes five departments of study – aerospace and mechanical engineering, chemical and biomolecular engineering, civil engineering and geological sciences, computer science and engineering, and electrical engineering – with eight B.S. degrees offered. Additionally, the college offers five-year dual degree programs with the Colleges of Arts and Letters and of Business awarding additional B.A. and Master of Business Administration (MBA) degrees, respectively.\"\n",
```

Comment:
Should this be changed as follows?
```suggestion
    "input = \"Question: How many departments are within the Stinson-Remick Hall of Engineering? Context: The College of Engineering was established in 1920, however, early courses in civil and mechanical engineering were a part of the College of Science since the 1870s. Today the college, housed in the Fitzpatrick, Cushing, and Stinson-Remick Halls of Engineering, includes five departments of study – aerospace and mechanical engineering, chemical and biomolecular engineering, civil engineering and geological sciences, computer science and engineering, and electrical engineering – with eight B.S. degrees offered. Additionally, the college offers five-year dual degree programs with the Colleges of Arts and Letters and of Business awarding additional B.A. and Master of Business Administration (MBA) degrees, respectively.\"\n",
```